api/src/main/java/javax/enterprise/event/Event.java

```diff
@@ -100,6 +106,36 @@
 
     /**
      * <p>
+     * Fires an event asynchronously with the specified qualifiers and notifies synchronous and asynchronous observers.
+     * </p>
+     *
+     * @param event the event object
+     * @throws IllegalArgumentException if the runtime type of the event object contains a type variable
+     * @throws FireAsyncException   if a notified observer throws an exception, it will be caught and added
```

Comment:
Yes, I used a wrong term. I meant "completeExceptionally" instead of fail.

So if my understanding is correct then `fireAsync` never throws an exception (even though Javadoc says so). Instead, if any of the synchronous or asynchronous observers throws an exception then the resulting CompletionStage is completed exceptionally with FireAsyncException that wraps all the exceptions raised by observers. If no exception is thrown by observers then the resulting CompletionStage is completed normally with the event payload.

If this is a correct understanding then it should be specified in the spec and Javadoc more explicitly.